_1327/user_management/authentication.py

```diff
@@ -1,29 +1,35 @@
 from django.contrib.auth.models import Group
 from django.contrib.contenttypes.models import ContentType
-from guardian.shortcuts import get_perms
+from guardian.core import ObjectPermissionChecker
+from guardian.utils import get_anonymous_user
 
 
-class IPRangeAuthorizationBackend:
+class _1327AuthorizationBackend:
 
 	def authenticate(self, *args, **kwargs):
 		return None
 
 	def has_perm(self, user_obj, perm, obj=None):
 		if obj is None:
 			return False
-		if user_obj.is_anonymous():
-			group_name = user_obj._ip_range_group_name if hasattr(user_obj, '_ip_range_group_name') else None
-			if group_name:
-				perm = perm.split(".")
-				app = perm[0]
-				perm = perm[1]
-
-				# Only permissions from the same app as the object are valid.
-				content_type = ContentType.objects.get_for_model(obj)
-				if app != content_type.app_label:
-					return False
-
-				group = Group.objects.get(name=group_name)
-				perms = get_perms(group, obj)
-				return perm in perms
-		return False
+
+		app, perm = perm.split(".")
+
+		# Only permissions from the same app as the object are valid.
+		content_type = ContentType.objects.get_for_model(obj)
+		if app != content_type.app_label:
+			return False
+
+		group_name = user_obj._ip_range_group_name if hasattr(user_obj, '_ip_range_group_name') else None
+
+		has_perm = False
+		if user_obj.is_authenticated():
+			# user is not anonymous user and has no permissions on other elements
+			# --> we need to check the anonymous permissions again
+			check = ObjectPermissionChecker(get_anonymous_user())
+			has_perm = check.has_perm(perm, obj)
+		elif group_name:
```

Comment:
why `elif`? shouldn't this test be done even if the user is authenticated?
please do an early return above if the user has the permission so that we can skip this part.